### PR TITLE
Vector async slicing nap

### PR DESCRIPTION
**Pull request by kcpevey**
# Description

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

File: docs/naps/4-async-slicing.md [hunk 1]
```diff
@@ -223,6 +223,29 @@ It's important to understand what state is currently used for slicing in napari.
...
+- `Vectors`
+    - `_view_data`: `(M, 2, 2) array`:
+        The start point and projections of N vectors in 2D for vectors whose
+        start point is in the currently viewed slice. Subset of `data`
+    - `_view_indices`: `(1, M) array`:
+        indices for the M in view vectors (indices for subsetting `data`)
+    - `_view_alphas`: `(M,) or float`:
+        relative opacity for the M in view vectors
+    - `_view_faces`:  `(2M, 3) or (4M, 3) np.ndarray`:
+        indices of the _mesh_vertices that form the faces of the M in view vectors.
```
Comment (andy-sweet) [inline docs/naps/4-async-slicing.md]:
```suggestion
        indices of the `_mesh_vertices` that form the faces of the M in view vectors.
```

File: docs/naps/4-async-slicing.md [hunk 2]
```diff
@@ -223,6 +223,29 @@ It's important to understand what state is currently used for slicing in napari.
...
+        Shape is (2M, 2) for 2D and (4M, 2) for 3D. 
+		* Subset of `_mesh_triangles`
+    - `_view_vertices`: `(4M, 2) or (8M, 2) np.ndarray`:
+        the corner points for the M in view faces. Shape is (4M, 2) for 2D and (8M, 2) for 3D.
+		* Subset of `_mesh_vertices`
+	- `out_of_slice_display`: `bool`:
+        If True, renders vectors not just in central plane but also slightly out of slice
+        according to specified point marker size.
+
+	- Note: `_view_faces` and `view_vertices` require:
```
Comment (andy-sweet) [inline docs/naps/4-async-slicing.md]:
```suggestion
	- Note: `_view_faces` and `_view_vertices` require:
```

File: docs/naps/4-async-slicing.md [hunk 3]
```diff
@@ -223,6 +223,29 @@ It's important to understand what state is currently used for slicing in napari.
...
+    - `_view_alphas`: `(M,) or float`:
+        relative opacity for the M in view vectors
+    - `_view_faces`:  `(2M, 3) or (4M, 3) np.ndarray`:
+        indices of the _mesh_vertices that form the faces of the M in view vectors.
+        Shape is (2M, 2) for 2D and (4M, 2) for 3D. 
+		* Subset of `_mesh_triangles`
+    - `_view_vertices`: `(4M, 2) or (8M, 2) np.ndarray`:
+        the corner points for the M in view faces. Shape is (4M, 2) for 2D and (8M, 2) for 3D.
+		* Subset of `_mesh_vertices`
+	- `out_of_slice_display`: `bool`:
```
Comment (andy-sweet) [inline docs/naps/4-async-slicing.md]:
```suggestion
    - `out_of_slice_display`: `bool`:
```

File: docs/naps/4-async-slicing.md [hunk 4]
```diff
@@ -297,3 +320,64 @@ CC0+BY [^cc0-by].
 This document is dedicated to the public domain with the Creative Commons CC0
 license [^id3]. Attribution to this source is encouraged where appropriate, as per
 CC0+BY [^id4].
+
+## Related Technical Details
```
Comment (andy-sweet) [inline docs/naps/4-async-slicing.md:347]: Love these details. Happy to merge them for now. We can think about the best place for them later this week.

**Review by andy-sweet — CHANGES_REQUESTED** [review]
Looks great - I just had some minor suggestions around formatting.

Also, you should also add yourself as an author to the doc at the top. Email is optional.

**Comment by kcpevey** [comment]
Thanks @andy-sweet! I've made the requested changes. This is ready for review again.

**Review by andy-sweet — APPROVED** [review]
Looks great, will merge this now!

**Pull request merged**